Foundation/Block Enumeration/TWTBlockEnumeration.m

```diff
@@ -75,6 +77,47 @@ + (id)performCollectOnObject:(id <NSFastEnumeration>)object resultsCollectionCla
 }
 
 
++ (id)performDictionaryFlattenOnObject:(NSDictionary *)dictionary
+{
+    NSParameterAssert(dictionary);
+    
+    NSMutableDictionary *flattenedDictionary = [[NSMutableDictionary alloc] init];
+    
+    for (id<NSCopying> key in dictionary) {
+        id result = dictionary[key];
+        
+        if ([result respondsToSelector:@selector(objectForKey:)]) {
+            [flattenedDictionary addEntriesFromDictionary:[self performDictionaryFlattenOnObject:result]];
+        }
+        else {
+            [flattenedDictionary setObject:result forKey:key];
+        }
+    }
+    
+    return flattenedDictionary;
+}
+
+
++ (id)performCollectionFlattenOnObject:(id <NSFastEnumeration>)collection resultsCollectionClass:collectionClass
+{
+    NSParameterAssert(collection);
+    NSParameterAssert(collectionClass);
+    
+    id flattenedCollection = [[collectionClass alloc] init];
+
+    for (id element in collection) {
+        if ([element respondsToSelector:@selector(objectAtIndex:)]) {
+            [flattenedCollection addObjectsFromArray:[self performCollectionFlattenOnObject:element resultsCollectionClass:collectionClass]];
```

Comment:
The `respondsToSelector:` check looks for the one method in the `NSFastEnumeration` protocol.